src/libgit2/transports/ssh.c

```diff
@@ -748,23 +750,32 @@ static int check_certificate(
 	if (check_cb != NULL) {
 		git_cert_hostkey *cert_ptr = &cert;
 		git_error_state previous_error = {0};
+		char *host_and_port;
+
+		if (port == SSH_DEFAULT_PORT) {
+			host_and_port = strdup(host);
+		} else {
+			/* Maximal port number is 65535, which has 5 digits. */
+			size_t n = strlen(host) + 5 + 2;
+			host_and_port = malloc(n);
+			snprintf(host_and_port, n, "%s:%d", host, port);
+		}
```

Comment:
We should use our `git_str` class here, and we should avoid the unnecessary malloc. Something more like:

```
const char *host_ptr = host;
git_str host_and_port = GIT_STR_INIT;

if (port != SSH_DEFAULT_PORT) {
    git_str_printf(&host_and_port, "%s:%d", host, port);
    host_ptr = host_and_port.ptr;
}

// use host_ptr

git_str_free(&host_and_port);